src/m3/test/constraints.test.ts

```diff
@@ -0,0 +1,94 @@
+import {
+    assertEquals
+} from "https://deno.land/std@0.160.0/testing/asserts.ts"
+
+import {
+    Annotation,
+    Concept,
+    ConceptInterface,
+    Containment,
+    Metamodel,
+    Multiplicity,
+    Property,
+    Reference
+} from "../types.ts"
+import {issuesMetamodel} from "../constraints.ts"
+
+
+Deno.test("constraints (lioncore)", async (tctx) => {
+
+    await tctx.step("check that a ConceptInterface only has derived features", async () => {
+        const metamodel = new Metamodel("metamodel")
+        const conceptInterface = new ConceptInterface(metamodel, "conceptInterface")
+        metamodel.havingElements(conceptInterface)
+        const property = new Property(conceptInterface, "property", Multiplicity.Single)
+        conceptInterface.havingFeatures(property)
+
+        const issues = issuesMetamodel(metamodel)
+        assertEquals(issues.length, 1)
+        const {location, message} = issues[0]
+        assertEquals(location, conceptInterface)
+        assertEquals(message, `The features of a ConceptInterface must all be derived, but the following feature of metamodel.conceptInterface is not: property.`)
+    })
+
+    await tctx.step("check that an Annotation is not the type of a Containment", async () => {
+        const metamodel = new Metamodel("metamodel")
+        const annotation = new Annotation(metamodel, "@nnotation")
```

Comment:
Maybe we need rules for identifiers, and @ could be forbidden

Reply:
I'm generally pretty relaxed on this, as I think it's the job of a code generator to map strings to valid identifiers, but a couple of constraints should be there, like:

- Identifiers should be unique (within the namespace)
- An identifier should be a non-empty, non-whitespace-only string